src/parser/mod.rs

```diff
@@ -7379,6 +7380,31 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse an REPLACE statement
+    pub fn parse_replace(&mut self) -> Result<Statement, ParserError> {
+        if !dialect_of!(self is MySqlDialect) {
+            return parser_err!("unmatched dialect", self.peek_token().location);
+        }
+
+        let insert = &mut self.parse_insert().unwrap();
+        if let Statement::Insert {
+            replace_into,
+            priority,
+            ..
+        } = insert
+        {
+            if *priority == Some(MysqlInsertPriority::HighPriority) {
```

Comment:
I don't understand why this is checking for `HIGH_PRIORITY` -- if it is a semantic check https://github.com/sqlparser-rs/sqlparser-rs?tab=readme-ov-file#syntax-vs-semantics please remove the check

If you want to change the parser to only parse `HIGH_PRIORITY` at this location that would be ok too

Reply:
REPLACE INTO statement doesn't support to HighPriority. But insert statement supports. So, it seems ok.

Reply:
I think this qualifies as a SEMANTIC check rather than a syntactic check. Can you please remove it?

Reply:
Check was removed.